javalib/src/main/scala/java/net/URI.scala

```diff
@@ -41,19 +41,19 @@ final class URI(origStr: String) extends Serializable with Comparable[URI] {
     else fld(AbsHierPart)
   }.get
 
-  private val _authority = fld(AbsAuthority, RelAuthority)
+  private val _authority = fld(AbsAuthority, RelAuthority).filter(_ != "")
   private val _userInfo = fld(AbsUserInfo, RelUserInfo)
   private val _host = fld(AbsHost, RelHost)
   private val _port = fld(AbsPort, RelPort).fold(-1)(_.toInt)
 
   private val _path = {
-    if (_isAbsolute) {
-      if (_authority.isDefined) fld(AbsNetPath)
-      else fld(AbsAbsPath)
-    } else {
-      if (_authority.isDefined) fld(RelNetPath)
-      else fld(RelAbsPath) orElse fld(RelRelPath)
-    }
+    val useNetPath = fld(AbsAuthority, RelAuthority).isDefined
+    if (useNetPath)
+      fld(AbsNetPath, RelNetPath) orElse ("": js.UndefOr[String])
```

Comment:
Then this would be of type `String` instead of `UndefOr[String]`. Then I would either put it back in an `UndefOr[String]` or change `_path` to `String` which would need changes all around the file.

Additionally type inference seems to broken with that:
```scala
[error] /Users/nastucki/GitHub/scala-js/javalib/src/main/scala/java/net/URI.scala:52: error: type mismatch;
[error]  found   : Object
[error]  required: scala.scalajs.js.UndefOr[String]
[error]       fld(AbsNetPath, RelNetPath).getOrElse("")
```
It Might be related to #2095.